src/pg4.rs

```diff
@@ -0,0 +1,1029 @@
+//! | `uuid::Uuid`            | uuid(string)                                  | UUID                         |
+//! | `serde_json::Value`     | jsonb(list\<u8\>)                             | JSONB                        |
+//! | `serde::De/Serialize`   | jsonb(list\<u8\>)                             | JSONB                        |
+//! | `rust_decimal::Decimal` | decimal(string)                               | NUMERIC                      |
+//! | `postgres_range`        | range-int32(...), range-int64(...)            | INT4RANGE, INT8RANGE         |
+//! | lower/upper tuple       | range-decimal(...)                            | NUMERICRANGE                 |
+//! | `Vec<Option<...>>`      | array-int32(...), array-int64(...), array-str(...), array-decimal(...) | INT4[], INT8[], TEXT[], NUMERIC[] |
+//! | `pg4::Interval`         | interval(interval)                            | INTERVAL                     |
+
+/// An open connection to a PostgreSQL database.
+///
+/// # Examples
+///
+/// Load a set of rows from a local PostgreSQL database, and iterate over them.
+///
+/// ```no_run
+/// use spin_sdk::pg4::{Connection, Decode};
+///
+/// # fn main() -> anyhow::Result<()> {
+/// # let min_age = 0;
+/// let db = Connection::open("host=localhost user=postgres password=my_password dbname=mydb")?;
+///
+/// let query_result = db.query(
+///     "SELECT * FROM users WHERE age >= $1",
+///     &[min_age.into()]
+/// )?;
+///
+/// let name_index = query_result.columns.iter().position(|c| c.name == "name").unwrap();
+///
+/// for row in &query_result.rows {
+///     let name = String::decode(&row[name_index])?;
+///     println!("Found user {name}");
+/// }
+/// # Ok(())
+/// # }
+/// ```
+///
+/// Perform an aggregate (scalar) operation over a table. The result set
+/// contains a single column, with a single row.
+///
+/// ```no_run
+/// use spin_sdk::pg4::{Connection, Decode};
+///
+/// # fn main() -> anyhow::Result<()> {
+/// let db = Connection::open("host=localhost user=postgres password=my_password dbname=mydb")?;
+///
+/// let query_result = db.query("SELECT COUNT(*) FROM users", &[])?;
+///
+/// assert_eq!(1, query_result.columns.len());
+/// assert_eq!("count", query_result.columns[0].name);
+/// assert_eq!(1, query_result.rows.len());
+///
+/// let count = i64::decode(&query_result.rows[0][0])?;
+/// # Ok(())
+/// # }
+/// ```
+///
+/// Delete rows from a PostgreSQL table. This uses [Connection::execute()]
+/// instead of the `query` method.
+///
+/// ```no_run
+/// use spin_sdk::pg4::Connection;
+///
+/// # fn main() -> anyhow::Result<()> {
+/// let db = Connection::open("host=localhost user=postgres password=my_password dbname=mydb")?;
+///
+/// let rows_affected = db.execute(
+///     "DELETE FROM users WHERE name = $1",
+///     &["Baldrick".to_owned().into()]
+/// )?;
+/// # Ok(())
+/// # }
+/// ```
+#[doc(inline)]
+pub use super::wit::pg4::Connection;
```

Comment:
We extend the equivalent type in sqlite3: https://github.com/spinframework/spin-rust-sdk/pull/81/files#diff-5d6f50ac9a55cf5e3cd0473391b4195d9aed79c7bd7d2e896c6124b70e8a4416R184-R199

Do we not want the same thing here?